src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -857,6 +857,12 @@ void VM_Version::get_processor_features() {
   }
 #endif
 
+  // Check if processor has Intel Ecore
+  if (FLAG_IS_DEFAULT(DoEcoreOpt) && is_intel() && cpu_family() == 6 &&
+    (_model == 0x97 || _model == 0xAC || _model == 0xAF)) {
```

Comment:
Model 0x97 corresponds to ADL Hybrid Core converged ISA target where DoEcodeOpt enabling will depend on the core over which VM initialization thread executes.

Reply:
On hybrid platforms, it will be enabled on both Pcore and Ecore. Performance is essentially unchanged on Pcore